### ISSUE TITLE
Add option to require authentication to see public toots

### ISSUE THREAD

**Issue by polymerwitch** [TootCat/mastodon#41]
Currently, non authenticated users can see public toots on our server by visiting the url (linked by clicking the toot timestamp). This allows blocked users to view toots. These toots would still be visible on third party instances (for example if there was a thread with mentions of toot.cat, witches.town, and banned.space, and toot.cat had banned.space instance blocked but witches.town did not, then the toots would still be publicly visible on witches town). While this doesn't completely solve the issue of public toots being discoverable it does make them harder to discover.

One downside to this is that instances we like (such as witches.town) will not be able to view the public toots if they view the toot.cat page, but will be able to view them through witches.town.

* * * *

- [ ] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by polymerwitch** [comment]
should also block unauthenticated access to public api endpoints

**Comment by polymerwitch** [comment]
This commit does what I'm looking for https://github.com/phildini/mastodon/commit/6d3d10413ee2eeb5865102877367e7cf9736a5d1

**Comment by lynnco** [comment]
will it still block viewing for witches.town users if they're logged in ?

**Comment by lynnco** [comment]
also this seems like a good reason to work on programmatic instance banlist sharing

**Comment by polymerwitch** [comment]
witches.town users will be able to see the toots from toot.cat on witches.town, but they will not be able to see them on toot.cat

**Comment by polymerwitch** [comment]
I'm not sure if making the api endpoints auth only breaks federation.

**Comment by lynnco** [comment]
because you'd have to have a toot.cat login to auth with the toot.cat server, yea?

**Comment by polymerwitch** [comment]
yeah, but our toots are federated out to instances like witches.town. So, they could still see the toots that federated there
